master/cluster.go

```diff
@@ -1289,6 +1289,25 @@ func (c *Cluster) markDeleteVol(name, authKey string, force bool) (err error) {
 		return proto.ErrVolNotExists
 	}
 
+	vol.mpsLock.RLock()
+	volDentryCount := uint64(0)
+	for _, mp := range vol.MetaPartitions {
```

Comment:
your lock cost maybe too bigger, which may casue other request.
here, you can use the backup of vol.metapartitions, eg: 
`vol.mpsLock.RLock()
newMap := vol.cloneMetaPartitions.
vol.mpsLock.RUnLock()
`

Reply:
good suggestion ! i will fix soon